android/core/src/main/java/com/mongodb/stitch/android/core/Stitch.java

```diff
@@ -103,15 +111,18 @@ public static synchronized StitchAppClient getDefaultAppClient() {
    * @param clientAppId the client app id of the app client to get.
    * @return the app client associated with the client app id.
    */
-  public static synchronized StitchAppClient getAppClient(
+  public static StitchAppClient getAppClient(
       @Nonnull final String clientAppId
   ) {
     ensureInitialized();
-    if (!appClients.containsKey(clientAppId)) {
-      throw new IllegalStateException(
-          String.format("client for app '%s' has not yet been initialized", clientAppId));
+
+    synchronized (Stitch.class) {
+      if (!appClients.containsKey(clientAppId)) {
+        throw new IllegalStateException(
+            String.format("client for app '%s' has not yet been initialized", clientAppId));
+      }
+      return appClients.get(clientAppId);
```

Comment:
because the above line is checking ```if (!appClients.containsKey(clientAppId))```, if another thread performs a destructive operation in between that check and the return, a user could receive a null result, which is not our current contract for that method

Reply:
Would my suggestion above behave any differently though?